vumi/middleware/base.py

```diff
@@ -176,7 +177,19 @@ class MiddlewareStack(object):
     """
 
     def __init__(self, middlewares):
-        self.middlewares = middlewares
+        self.consume_middlewares = self._sort_by_priority(
+            middlewares, 'consume_priority')
+        self.publish_middlewares = self._sort_by_priority(
+            reversed(middlewares), 'publish_priority')
+
+    @staticmethod
+    def _sort_by_priority(middlewares, priority_key):
+        priorities = defaultdict(list)
+        for mw in middlewares:
+            priorities[mw.config.get(priority_key, None)].append(mw)
+        sorted_priorities = sorted(
+            priorities.keys(), key=lambda a: '' if not a else a)
+        return [mw for p in sorted_priorities for mw in priorities[p]]
```

Comment:
I wonder if we could just replace this function with ``sorted(middlewares, key=lambda mw: mw.config.get(priority_key, ''))`` and rely on Python's sort being a stable sort?